src/common/mod.rs

```diff
@@ -107,40 +107,22 @@ impl CoseError {
     }
 }
 
-/// Newtype wrapper around a byte slice to allow left-over data to be detected.
-struct MeasuringReader<'a>(&'a [u8]);
-
-impl<'a> MeasuringReader<'a> {
-    fn new(buf: &'a [u8]) -> MeasuringReader<'a> {
-        MeasuringReader(buf)
-    }
-
-    fn is_empty(&self) -> bool {
-        self.0.is_empty()
-    }
-}
-
-impl<'a> ciborium_io::Read for &mut MeasuringReader<'a> {
-    type Error = EndOfFile;
-
-    fn read_exact(&mut self, data: &mut [u8]) -> Result<(), Self::Error> {
-        if data.len() > self.0.len() {
-            return Err(EndOfFile);
-        }
-
-        let (prefix, suffix) = self.0.split_at(data.len());
-        data.copy_from_slice(prefix);
-        self.0 = suffix;
-        Ok(())
+/// Convert a Ciborium error to one that uses our EndOfFile marker.
+fn convert_cbor_err<T>(e: cbor::de::Error<T>) -> cbor::de::Error<EndOfFile> {
+    use cbor::de::Error::{Io, RecursionLimitExceeded, Semantic, Syntax};
+    match e {
```

Comment:
I did consider adding a new From, but I didn't want to do that because then it would become public, in the same way the existing From impls show up in the docs and can be accessed. And with your suggestion we would be changing one, with potential backward compatibility issues? But if that's not a big deal, that seems better.

Reply:
Adding it to the public API seems OK to me, and I *think* it [doesn't count as a breaking change](https://rust-lang.github.io/rfcs/1105-api-evolution.html#minor-change-implementing-any-non-fundamental-trait).

Reply:
OK, done.